src/components/Registration/index.jsx

```diff
@@ -0,0 +1,126 @@
+      setPasswordConfirmation(e.target.value)
+    }
+
+    const changefullnameHandler = (e) => {
+      setFullnameHandler(e.target.value)
+    }
+
+    const handlePasswordToggle = () => {
+      if(showPassword){
+        setShowPassword(false)
+      } else {
+        setShowPassword(true)
+      }
+    }
+
+    const handlePasswordConfirmationToggle = () => {
+      if(showPasswordConfirmation){
+        setShowPasswordConfirmation(false)
+      } else {
+        setShowPasswordConfirmation(true)
+      }
+    }
+
+    return (
+      <div className="container-registration">
+        <img src={IMAGES.union} alt="Black market logo" />
+        <form onSubmit={submitHandler}>
+          <div className="email">
+            <label htmlFor="email">Email</label>
+            <input
+              type="email"
+              placeholder="Type your email"
+              onChange={changeEmailHandler} value={email}
+              required
+            />
+          </div>
+          <div className="full-name">
+            <label htmlFor="fullname">Fullname</label>
+            <input
+              type="fullname"
+              placeholder="Type your full name"
+              onChange={changefullnameHandler} value={fullname}
+              required
+            />
+          </div>
+          <div className="password">
+            <label htmlFor="password">Password</label>
+            <input
+              type={showPassword ? 'text' : 'password'}
+              placeholder="Type your password"
+              onChange={changePasswordHandler} value={password}
+              required
+            />
+            <div className="eye-icon" onClick={handlePasswordToggle} title="Show password">
+              <Eye />
+            </div>
+          </div>
+          <div className="password-confirmation">
+            <label htmlFor="password-confirmation">Confirm Password</label>
+            <input
+              type={showPasswordConfirmation ? 'text' : 'password'}
+              placeholder="Type your password"
+              onChange={changePasswordConfirmationHandler} value={password_confirmation}
+              required
+            />
+            <div className="eye-icon" onClick={handlePasswordConfirmationToggle} title="Show password">
+              <Eye />
+            </div>
+          </div>
+          <button className="submit" disabled={!email || !fullname || !password || !password_confirmation}>
+            Sign up
+          </button>
+        </form>
+        <div className="error-messages">
+          { registrationState.error && <strong>{registrationState.message}</strong> }
+        </div>
+        <div className="policies">
+          <p>By signing up, you accept the <Link to="/">Data Policy</Link> and the <Link to="/">Cookies Policy</Link>.</p>
+        </div>
+        <div className="account">
+          <p>Already have an account? <Link to="/login">Log in</Link></p>
+        </div>
+      </div>
+    )
+  }
```

Comment:
This component is super similar to the one you made for login, you can check which parts you can reuse and create common components, like inputs, buttons or even the form